### PR TITLE
chore: improve consensus/execution layer clients logging and error handling

### PR DESCRIPTION
**Pull request by olegshmuelov**
close https://github.com/ssvlabs/ssv/issues/1783

### CODE REVIEW

File: beacon/goclient/proposer.go [hunk 1]
```diff
@@ -58,18 +65,29 @@ func (gc *GoClient) GetBeaconBlock(slot phase0.Slot, graffitiBytes, randao []byt
...
+		gc.log.Error(clNilResponseDataErrMsg,
+			zap.String("api", "Proposal"),
+		)
 		return nil, DataVersionNil, fmt.Errorf("proposal data is nil")
 	}
 
 	metricsProposerDataRequest.Observe(time.Since(reqStart).Seconds())
 	beaconBlock := proposalResp.Data
 
+	// TODO: consider adding logger.Error here as well
```
Comment (olegshmuelov) [inline beacon/goclient/proposer.go:90]: we have to decide if we want to add err logging here as well

File: protocol/v2/ssv/runner/committee.go [hunk 2]
```diff
@@ -445,6 +446,7 @@ func (cr *CommitteeRunner) ProcessPostConsensus(ctx context.Context, logger *zap
 	if len(attestations) > 0 {
 		submissionStart := time.Now()
 		if err := cr.beacon.SubmitAttestations(attestations); err != nil {
+			// TODO: consider removing the logger.Error here
```
Comment (olegshmuelov) [inline protocol/v2/ssv/runner/committee.go:449]: duplicate err. we should consider removing it

Comment (y0sher) [reply protocol/v2/ssv/runner/committee.go:449]: since this is in the context of the runner I think we can leave it even though its a duplicate.

**Review by y0sher — APPROVED** [review]
gj

File: protocol/v2/ssv/runner/committee.go (hunk 2 above)
Comment (moshe-blox) [reply protocol/v2/ssv/runner/committee.go:449]: yeah because you wont see the Beacon logs when filtering for some duty

**Pull request merged**